stacks/services/APIGateway.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    AuthorizationType,
+    AwsIntegration,
+    MethodOptions,
+    PassthroughBehavior,
+    Resource,
+} from "aws-cdk-lib/aws-apigateway";
+import { PolicyStatement, Role, ServicePrincipal } from "aws-cdk-lib/aws-iam";
+import { ApiGatewayV1Api, Stack, Bucket } from "sst/constructs";
+
+export const createSiriApi = (stack: Stack, siriSXBucket: Bucket): void => {
+    const apiGateway = new ApiGatewayV1Api(stack, "cdd-siri-sx-api", {
+        cdk: {
+            restApi: {
+                restApiName: `cdd-siri-sx-api-${stack.stage}`,
+                description: "API to retrieve Siri SX XML data",
+            },
+        },
+    });
+
+    const bucketResource: Resource = apiGateway.cdk.restApi.root.addResource("{folder}");
+    const objectResource: Resource = bucketResource.addResource("{item}");
+
+    const executeRole = new Role(stack, "cdd-siri-sx-api-gateway-s3-integration-role", {
+        assumedBy: new ServicePrincipal("apigateway.amazonaws.com"),
+        roleName: `cdd-siri-sx-api-gateway-s3-integration-role-${stack.stage}`,
+    });
+
+    executeRole.addToPolicy(
+        new PolicyStatement({
+            resources: [`${siriSXBucket.bucketArn}/*`],
+            actions: ["s3:GetObject"],
+        }),
+    );
+
+    const s3Integration: AwsIntegration = new AwsIntegration({
+        service: "s3",
+        region: "eu-west-2",
+        integrationHttpMethod: "GET",
+        path: "{bucket}/{object}",
+        options: {
+            credentialsRole: executeRole,
+            passthroughBehavior: PassthroughBehavior.WHEN_NO_MATCH,
+            requestParameters: {
```

Comment:
So these requestParameters can go and the path could just be `${siriSxBucket.bucketName}/SIRI-SX.xml`

Reply:
done